### DISCUSSION TITLE
Unable to push metrics to Datadog

### DISCUSSION THREAD

**Discussion by okHadi** [artilleryio/artillery#3347] [Artillery]
This is my current config.yml:

```config:
  target: "https://www.example.com"
  phases:
    - duration: 30 
      arrivalCount: 10
  config:
    plugins:
    publish-metrics:
      - type: datadog
        # DD_API_KEY is an environment variable containing the API key
        apiKey: "{{ $env.DD_API_KEY }}"
        apiHost: app.datadoghq.eu
        tags:
          - "testId:{{ $testId }}"
          - "reporterType:datadog-api"

scenarios:
  - flow:
      - get:
          url: "/"
      - log: "API key is: {{ $env.DD_API_KEY }}"
```

The API Key logs fine. It is a very basic test and I expect some metrics to be available in Datadog. However, I see none. I've tried running locally as well as on GitHub Actions but I see no metrics. I've to replicate this simple POC for a bigger loadtest but I can't get it this to work.

Testing Github Actions workflow: https://github.com/okHadi/dd-artillery/actions/runs/11015991845

I've tested this on two different systems with two different accounts but still metrics are not being published.

**Comment by hassy** [comment]
The `plugins` section does not need to be nested under its own `config` section - opened a PR here: https://github.com/okHadi/dd-artillery/pull/1

**Comment by okHadi** [comment]
Thank you so much @hassy , that fixed it.
Just curious; why doesn't artillery throw an error if there is a mistake like this in the config.yml?
